dcos/dcos-cluster-one-master.json

```diff
@@ -699,48 +744,36 @@
       "type": "Microsoft.Storage/storageAccounts"
     }, 
     {
-      "apiVersion": "[variables('apiVersionDefault')]", 
+      "apiVersion": "2016-07-01",
+      "name": "[variables('virtualNetworkName')]",
+      "type": "Microsoft.Resources/deployments", 
       "dependsOn": [
         "[concat('Microsoft.Network/networkSecurityGroups/', variables('agentPublicNSGName'))]", 
         "[concat('Microsoft.Network/networkSecurityGroups/', variables('agentPrivateNSGName'))]"
       ], 
-      "location": "[resourceGroup().location]", 
-      "name": "[variables('virtualNetworkName')]", 
-      "properties": {
-        "addressSpace": {
-          "addressPrefixes": [
-            "[variables('masterAddressPrefix')]", 
-            "[variables('agentAddressPrefix')]"
-          ]
-        }, 
-        "subnets": [
-          {
-            "name": "[variables('masterSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('masterAddressPrefix')]"
-            }
-          }, 
-          {
-            "name": "[variables('agentPublicSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('agentPublicAddressPrefix')]", 
-              "networkSecurityGroup": {
-                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('agentPublicNSGName'))]"
-              }
-            }
-          }, 
-          {
-            "name": "[variables('agentPrivateSubnetName')]", 
-            "properties": {
-              "addressPrefix": "[variables('agentPrivateAddressPrefix')]", 
-              "networkSecurityGroup": {
-                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('agentPrivateNSGName'))]"
-              }
-            }
-          }
-        ]
-      }, 
-      "type": "Microsoft.Network/virtualNetworks"
+      "properties": { 
+        "mode": "incremental", 
+        "templateLink": {
+          "uri": "[concat('https://raw.githubusercontent.com/brendandburns/acs/master/dcos/dcos-', parameters('virtualNetworkNewOrExisting'), '-vnet.json')]",
```

Comment:
1) Yeah, this is a temporary state of affairs,as it's sort of a circular dependency. ARM needs a working URL in order to download the other template, so this is pointed at mine for now.  Once it's merged I'll send a quick follow up PR to point it the main repo.

2) The only way to do conditional evaluation (e.g. "create in existing vnet" vs "create a new vnet") in ARM is to have sub-templates.  I don't think that we want to the combinatorial explosion of `dcos-one-master-with-vnent` vs `dcos-one-master-new-vnet` etc.  (esp. since it will multiply every time I have a new feature to turn on/off)

I'm ok with some sort of modularity + generation, but that's not where we are right now, so I'd say we get this merged and then do something comprehensive around tools, if that's the direction we want to go.